src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -35,16 +35,22 @@ async function getQuote({
   let exchangeAmount
   if (kind === 'sell') {
     // Sell orders need to deduct the fee from the swapped amount
-    exchangeAmount = BigNumber.from(amount)
-      .sub((await feePromise).amount)
-      .toString()
+    // we need to check for 0/negative exchangeAmount should fee >= amount
+    const { amount: fee } = await feePromise
+    const result = BigNumber.from(amount).sub(fee)
+
+    const validQuery = result.gt('0')
```

Comment:
can you give a more semantic name? --> i.e `positiveExchangeAmount`

Reply:
O use the name you used in the other method `feeExceedsPrice` and actually you can even return this, so you can use it directly (instead of relying on assuming that a null price is because `feeExceedsPrice`

Reply:
true, good points

Reply:
https://github.com/gnosis/gp-swap-ui/pull/645/commits/4d43b2dd7c4b2c94b2a22a9ca212cd80838ad1e6